src/main/java/GeneralService/Configurations.java

```diff
@@ -17,15 +17,15 @@
 @Configuration
 public class Configurations {
     private static final Logger LOGGER = LogManager.getLogger(Configurations.class.getName());
-
+    private final CustomFileReaderFactory fileReaderFactory = new CustomFileReaderFactory();
```

Comment:
delete this, should call the factory when you need it, no point in assigning it to a variable